server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -699,12 +708,29 @@ public static class CheckpointState implements Writeable {
          */
         boolean replicated;
 
+        /**
+         * The currently searchable replication checkpoint.
+         */
+        ReplicationCheckpoint visibleReplicationCheckpoint;
+
+        /**
+         * Map of ReplicationCheckpoints to ReplicationTimers.  Timers are added as new checkpoints are published, and removed when
+         * the replica is caught up.
+         */
+        Map<ReplicationCheckpoint, ReplicationTimer> checkpointTimers;
```

Comment:
I went back/forth on this.  The benefit of handing off is that our pressure settings are easier to reason about as they would be honored across handoffs, meaning pressure and eventually failure of a stale replica would be applied earlier on a newly elected primary post handoff.

However, we currently don't pause & resume replications, so if a handoff cuts a replica's sync to the old primary the replica would need to start over from the new primary.  I felt this could be too aggressive in rejections, and lead to unnecessary failures post relocation.  

I chose to keep it simple and not handoff with this PR, meaning pressure settings & timers represent the replica's state relative to the current primary, giving relocation the opportunity to recompute times & catch up any stale replica.  This also makes more sense with the addition to the nodestats API, as the results are primary/node specific.

Curious what you think here.

Reply:
Thanks @mch2  for explaining this. Yes, let's start with simple, we can always iterate over it based on feature feedback.